seed/views/v3/users.py

```diff
@@ -401,7 +401,7 @@ def update(self, request, pk=None):
         user.first_name = json_user.get('first_name')
         user.last_name = json_user.get('last_name')
         user.email = json_user.get('email')
-        user.username = json_user.get('email')
+        user.username = json_user.get('email', '').lower()
```

Comment:
No worries, I've been trying to dig around carefully myself! The issue I'm solving is one where a user told me she had updated her information and then wasn't able to log in. I've noticed that in SEED, user email addresses are case-sensitive and usernames are always pulled from the email address and then lowercased. This way the input username can be [lowercased for the case-sensitive authentication](https://github.com/SEED-platform/seed/blob/develop/seed/landing/views.py#L44) during login.

So with this user, because her email address is ExampleAddress@a.com, the above line in `update` had changed her username and email from `exampleaddress@a.com` and `ExampleAddress@a.com` to both being `ExampleAddress@a.com`. So because the email address she tries to log in with will get lowercased and compared case-sensitively to `ExampleAddress@a.com`, she'd never be able to log in.

A different issue is that, you're right, I'm not sure users should be able to change their email addresses (and by extension usernames) so easily. Especially since we in BEAM use email addresses for building lookups. I'd like to add something where some kind of verification email is sent out after updating an email address, but that's really only one piece to this.

Reply:
yes, we should always be using `.lower()` this has been an issue for quite some time and I think we have forces most emails to be `.lower()`

Reply:
so we are good to push? should we and a constraint and a migration?